jobs/webcompat-kb/webcompat_kb/main.py

```diff
@@ -569,10 +590,70 @@ def update_history(self, records):
 
         return records
 
-    def filter_bug_history_changes(self, history):
-        res = []
+    def get_existing_history_records_by_ids(self, bug_ids):
+        numbers_list = list(bug_ids)
+        formatted_numbers = ", ".join(map(str, numbers_list))
+
+        query = f"""
+                    SELECT *
+                    FROM `{self.bq_dataset_id}.bugs_history`
+                    WHERE number IN ({formatted_numbers})
+                """
+        result = self.client.query(query).result()
+        return result
+
+    def filter_only_unsaved_changes(self, history_updates, bug_ids):
+        existing_records = self.get_existing_history_records_by_ids(bug_ids)
+
+        # Return history_updates updates early if no saved records were found
+        if not existing_records:
+            return history_updates
 
-        for bug_history in history:
+        stored = defaultdict(list)
+
+        for row in existing_records:
+            row_dict = dict(row.items())
+            stored[row_dict["number"]].append(row_dict)
+
+        history_from_bq = dict(stored)
+
+        history_bq_set = set()
+        for number, changes in history_from_bq.items():
```

Comment:
Seems like we don't ever actually use `number` inside the loop.

Can we in fact just construct a set of all updates in bigquery and a set of all updates in bugzilla and find the difference i.e. those changes not in bigquery but in bugzilla?